applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/prebuckling_strategy.hpp

```diff
@@ -0,0 +1,980 @@
+
+            if (pScheme->SchemeIsInitialized() == false)
+                pScheme->Initialize(rModelPart);
+
+            if (pScheme->ElementsAreInitialized() == false)
+                pScheme->InitializeElements(rModelPart);
+
+            if (pScheme->ConditionsAreInitialized() == false)
+                pScheme->InitializeConditions(rModelPart);
+        }
+        //initialisation of the convergence criteria
+        pConvergenceCriteria->Initialize(BaseType::GetModelPart());
+
+        this->SetIsInitialized( true );
+
+        KRATOS_INFO_IF("PrebucklingStrategy", BaseType::GetEchoLevel() > 2 && rank == 0)
+            << "Exiting Initialize" << std::endl;
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * Clears the internal storage
+     */
+    void Clear() override
+    {
+        KRATOS_TRY
+
+        BuilderAndSolverPointerType &pBuilderAndSolver = this->pGetBuilderAndSolver();
+        pBuilderAndSolver->GetLinearSystemSolver()->Clear();
+        BuilderAndSolverPointerType &pEigenSolver = this->pGetEigenSolver();
+        pEigenSolver->GetLinearSystemSolver()->Clear();
+
+        // if (this->pGetMassMatrix() != nullptr)
+        //     this->pGetMassMatrix() = nullptr;
+
+        if (this->pGetStiffnessMatrix() != nullptr)
+            this->pGetStiffnessMatrix() = nullptr;
+
+        if (this->pGetStiffnessMatrixPrevious() != nullptr)
+            this->pGetStiffnessMatrixPrevious() = nullptr;
+
+        if (this->pGetRHS() != nullptr)
+            this->pGetRHS() = nullptr;
+
+        if (this->pGetDx() != nullptr)
+            this->pGetDx() = nullptr;
+
+        // Re-setting internal flag to ensure that the dof sets are recalculated
+        pBuilderAndSolver->SetDofSetIsInitializedFlag(false);
+        pEigenSolver->SetDofSetIsInitializedFlag(false);
+
+        pBuilderAndSolver->Clear();
+        pEigenSolver->Clear();
+
+        this->pGetScheme()->Clear();
+
+        mInitializeWasPerformed = false;
+        mSolutionStepIsInitialized = false;
+
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * Performs all the required operations that should be done (for each step)
+     * before solving the solution step.
+     * A member variable should be used as a flag to make sure this function is called only once per step.
+     */
+    void InitializeSolutionStep() override
+    {
+        KRATOS_TRY
+        if (!mSolutionStepIsInitialized){
+            ModelPart &rModelPart = BaseType::GetModelPart();
+            const int rank = rModelPart.GetCommunicator().MyPID();
+
+            KRATOS_INFO_IF("PrebucklingStrategy", BaseType::GetEchoLevel() > 2 && rank == 0)
+                << "Entering InitializeSolutionStep" << std::endl;
+
+            // Solver
+            BuilderAndSolverPointerType &pEigenSolver = this->pGetEigenSolver();
+            BuilderAndSolverPointerType &pBuilderAndSolver = this->pGetBuilderAndSolver();
+            // Scheme
+            SchemePointerType &pScheme = this->pGetScheme();
```

Comment:
Pointers does not require &, besides I prefer the & in the side of the type. That's only my opinion but I think is in the style guide

Reply:
@philbucher

Reply:
If I remove the & for the Member Variables like StiffnessMatrix, I get an segmentation fault.